src/js/actions/transform.js

```diff
@@ -524,6 +476,17 @@ define(function (require, exports) {
             sizePromise = layerActionsUtil.playLayerActions(document, resizeLayerActions, true, options);
         }
 
+        // Reset type layers to pick up their implicit font size changes
+        sizePromise = sizePromise
+            .bind(this)
+            .then(function () {
+                var typeLayers = layerSpec.filter(function (layer) {
+                    return layer.kind === layer.layerKinds.TEXT;
+                });
+
+                return this.transfer(layerActions.resetLayers, document, typeLayers);
```

Comment:
I understand the issue now. Your suggestion makes perfect sense. Thank you! I've amended the PR and force-pushed a replacement commit.